### PR TITLE
Ignore test inputs/outputs for Github Linguist

### PR DESCRIPTION
**Pull request by kulp**
I noticed that Github interprets `rust-bindgen` as a C++ project, due to the many C++ input headers. I find this misleading and undesirable.

screen shot from pinned repository

The use of `.gitattributes` for Github's Linguist is described in [`linguist/docs/overrides.md`](https://github.com/github/linguist/blob/02ffab2d4df3034c05fd7b2e8bf0a2fd31c0452d/docs/overrides.md).

The changes in this PR are certainly debatable; if they are not loved, I am happy to withdraw them.

### CODE REVIEW

**Comment by kulp** [comment]
This PR is a draft until I can confirm somehow that I got the syntax correct.

File: tests/headers/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Tell Github Linguist to avoid counting these C++ test inputs toward
+# statistics.
+*.hpp -linguist-detectable
```
Comment (emilio) [inline tests/headers/.gitattributes:3]: Also probably `.h`, right?

Comment (kulp) [reply tests/headers/.gitattributes:4]: Oh yes, thanks.

**Comment by kulp** [comment]
My initial attempts (such as 380bd83ee27c1cc14de2deda3b3fbe810b7ccd2a) failed to account for some files, mainly the enormous `stylo.hpp`.

After [installing](https://github.com/github/linguist#installation) and using `github-linguist --breakdown` [at the command line](https://github.com/github/linguist#command-line-usage), I see that the changes have taken effect:
```
97.55%  1046954    Rust
1.61%   17318      Python
0.65%   6953       Shell
0.14%   1532       Batchfile
0.04%   418        C
0.01%   102        Makefile

Rust:
bindgen-integration/build.rs
bindgen-integration/src/lib.rs
build.rs
src/callbacks.rs
src/clang.rs
src/codegen/bitfield_unit.rs
src/codegen/bitfield_unit_tests.rs
src/codegen/dyngen.rs
src/codegen/error.rs
src/codegen/helpers.rs
src/codegen/impl_debug.rs
src/codegen/impl_partialeq.rs
src/codegen/mod.rs
src/codegen/struct_layout.rs
src/deps.rs
src/extra_assertions.rs
src/features.rs
src/ir/analysis/derive.rs
src/ir/analysis/has_destructor.rs
src/ir/analysis/has_float.rs
src/ir/analysis/has_type_param_in_array.rs
src/ir/analysis/has_vtable.rs
src/ir/analysis/mod.rs
src/ir/analysis/sizedness.rs
src/ir/analysis/template_params.rs
src/ir/annotations.rs
src/ir/comment.rs
src/ir/comp.rs
src/ir/context.rs
src/ir/derive.rs
src/ir/dot.rs
src/ir/enum_ty.rs
src/ir/function.rs
src/ir/int.rs
src/ir/item.rs
src/ir/item_kind.rs
src/ir/layout.rs
src/ir/mod.rs
src/ir/module.rs
src/ir/objc.rs
src/ir/template.rs
src/ir/traversal.rs
src/ir/ty.rs
src/ir/var.rs
src/lib.rs
src/log_stubs.rs
src/main.rs
src/options.rs
src/parse.rs
src/regex_set.rs
src/time.rs
tests/expectations/build.rs
tests/expectations/lib.rs
tests/parse_callbacks/mod.rs
tests/quickchecking/src/bin.rs
tests/quickchecking/src/fuzzers.rs
tests/quickchecking/src/lib.rs
tests/quickchecking/tests/fuzzed-c-headers.rs
tests/stylo_sanity.rs
tests/tests.rs

Batchfile:
ci/assert-no-diff.bat
ci/test.bat

Shell:
ci/no-includes.sh
ci/test.sh
releases/friends.sh
tests/test-one.sh

C:
csmith-fuzzing/csmith.h

Python:
csmith-fuzzing/driver.py
csmith-fuzzing/predicate.py

Makefile:
tests/expectations/tests/enum-default-rust.d

```

**Review by emilio — APPROVED** [review]
Thank you!

**Comment by kulp** [comment]
Much better (at least to me) :)
Screen Shot 2022-06-05 at 17 40 55
Screen Shot 2022-06-05 at 17 41 12

**Pull request merged**